HSEARCH-4099 Allow option for setting Elasticsearch connection information via full URIs (#2442)
hibernate/hibernate-search: documentation/src/main/asciidoc/reference/backend-elasticsearch.asciidoc

This property may be set to either `http` or `https`.

==== Provide complete URIs

> Alternatively to `hosts` and `protocol`, it is possible to define both the terms using the same property:

Reviewer: ```suggestion
 Alternatively, it is possible to define both the protocol and hosts as one or more URIs using a single property:
```

[source]
----
hibernate.search.backend.uris = http://localhost:9200
----

This property may be set to a String representing an URI such as `http://localhost` or `https://es.mycompany.com:4400`,
or a String containing multiple such URI strings separated by commas,
or a `Collection<String>` containing such URI strings.

[WARNING]
====

> Note that there atrre some constraints we need to respect in order to use this property:

Reviewer: ```suggestion
There are some constraints regarding the use of this property:
```

* All the uris must have the same protocol.
* Cannot be used if `hosts` or `protocol` are set.

> * It must contain at least a valid URI.

Reviewer: ```suggestion
* The provided list of URIs must not be empty.
```
